llvm/lib/Target/NVPTX/NVPTXIntrinsics.td

```diff
@@ -2381,29 +2387,41 @@ def INT_PTX_LDG_G_v4i32_ELE : VLDG_G_ELE_V4<"u32", Int32Regs>;
 def INT_PTX_LDG_G_v4f32_ELE : VLDG_G_ELE_V4<"f32", Float32Regs>;
 
 
-multiclass NG_TO_G<string Str> {
+multiclass NG_TO_G<string Str, list<Predicate> Preds = []> {
    def "" : NVPTXInst<(outs Int32Regs:$result), (ins Int32Regs:$src),
-          "cvta." # Str # ".u32 \t$result, $src;", []>;
+          "cvta." # Str # ".u32 \t$result, $src;", []>, Requires<Preds>;
+   def _64 : NVPTXInst<(outs Int64Regs:$result), (ins Int64Regs:$src),
+          "cvta." # Str # ".u64 \t$result, $src;", []>, Requires<Preds>;
+}
+
+multiclass NG_TO_G_64<string Str, list<Predicate> Preds = []> {
    def _64 : NVPTXInst<(outs Int64Regs:$result), (ins Int64Regs:$src),
-          "cvta." # Str # ".u64 \t$result, $src;", []>;
+          "cvta." # Str # ".u64 \t$result, $src;", []>, Requires<Preds>;
 }
```

Comment:
I think it would be cleaner to just add a bit to the `NG_TO_G` class for `supports_32`